### PR TITLE
[SYCL][E2E] Track amount of `XFAIL`ed tests

### PR DESCRIPTION
**Pull request by AlexeySachkov**
We have a problem with our approach to `XFAIL`ing failed tests - we don't always submit trackers for those failures, thus simply hiding them from us.

Some of tests we `XFAIL`ed have been in that state for years and without a tracker submitted about them, we don't even know that there are issues.

This patch introduces a new requirement for marking test as expected to fail: every `XFAIL` directive has to be followed by `XFAIL-TRACKER` which points to intel/llvm issue submitted to analyze the failure and remove `XFAIL` directive.

The tracker can be referenced either by URL, or through GH shortcuts `owner/repo#NNNNN`.

To ensure that the new requirement is followed, a new test was added which checks amount of improper `XFAIL` directives (i.e. those which are not followed by `XFAIL-TRACKER` directive).

Similar approach is expected to be applied later for `UNSUPPORTED` directives, but it will be done as a separate PR.

### CODE REVIEW

File: sycl/test-e2e/README.md
Comment (uditagarwal97) [inline sycl/test-e2e/README.md:324]: Should we do the same for the tests marked "UNSUPPORTED"?

Comment (bader) [reply sycl/test-e2e/README.md:324]: > Should we do the same for the tests marked "UNSUPPORTED"?

@uditagarwal97, the last sentence in the description answers this question. ;)

Comment (uditagarwal97) [reply sycl/test-e2e/README.md:324]: Ahh.. I overlooked that. Time for me to get some coffee ☕

File: sycl/test-e2e/no-xfail-without-tracker.cpp
Comment (maarquitos14) [inline sycl/test-e2e/no-xfail-without-tracker.cpp:26]:
```suggestion
// The number below is a number of tests which are *improperly* XFAIL-ed, i.e.
```

File: sycl/test-e2e/no-xfail-without-tracker.cpp
Comment (maarquitos14) [inline sycl/test-e2e/no-xfail-without-tracker.cpp:33]:
```suggestion
// That number *must not* increase. Any PR which causes this number to group
```

Is `group` a typo? I guess you meant `grow`?

File: sycl/test-e2e/no-xfail-without-tracker.cpp
Comment (maarquitos14) [inline sycl/test-e2e/no-xfail-without-tracker.cpp:41]:
```suggestion
// tests to match the required format and in that case you should just update
```

**Review by maarquitos14 — APPROVED** [review]

File: sycl/test-e2e/no-xfail-without-tracker.cpp
Comment (AlexeySachkov) [reply sycl/test-e2e/no-xfail-without-tracker.cpp:33]: Right, I type `group` way more often than `grow`, so my hands betray me here :)

File: sycl/test-e2e/README.md
Comment (KornevNikita) [inline sycl/test-e2e/README.md:338]: Just wondering - can't we proceed with the full link instead? Looks like I need to go to the "issues", open some and then paste the number to find the ticket (is there another way?)

Comment (AlexeySachkov) [reply sycl/test-e2e/README.md:338]: The format allows full links here and, it is just that example uses a shortcut. I agree that it will be easier to have a full link, so let me make it stricter

Comment (AlexeySachkov) [reply sycl/test-e2e/README.md]: c908d656b6bcfd021466beb4c5d5a3ff8f5a91c7

**Comment by AlexeySachkov** [comment]
Local merge highlighted that amount of `XFAIL`ed test changed, hence the update to the PR. I'm not entirely sure if it worth waiting for CI to complete, because I've verified the test locally and other patches with `XFAIL`s could be merged while we are waiting for CI results here. Considering that the new test is only executed on Linux, I will cancel Windows workflows

**Comment by AlexeySachkov** [comment]
Pre-commit passed, I just did a local merge merge with `sycl` and the test still passed, so I will proceed with merge of the PR. There could be some failures in the new test after I merge this - simply because someone passed pre-commit without this patch, but they should be addressed as a usual bugfix.

**Pull request merged**
